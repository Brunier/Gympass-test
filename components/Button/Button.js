import React from 'react';
import styled from "styled-components";


const ButtonStyle = styled.button`
    padding: 16px 20px 14px 20px;
    background: ${props => props.disabled ? "#ccc" : "#00757E"};
    border: none;
    color: #fff;
    line-height: 21px;
    font-size: 20px;
    cursor: pointer;
 `;

export class Button extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <ButtonStyle type={this.props.type} disabled={this.props.disabled} onClick={this.props.onClick}>{this.props.text}</ButtonStyle>
        )
    }
}

export default Button;
