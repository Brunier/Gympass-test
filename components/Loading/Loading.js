import React from 'react';
import styled from "styled-components";

const Container = styled.div`
    text-align: center;
    display: ${props => props.display ? props.display : "block"};
    vertical-align: middle;
    position: relative;
    left: ${props => props.left ? props.left : "0"};
    top: 5px;
  
  img {
    width: ${props => props.width + "px"};
  }
 `;

export class Loading extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <Container {...this.props}>
                <img src={this.props.img ? this.props.img : "/static/loading.gif"} />
            </Container>
        )
    }
}

export default Loading;
