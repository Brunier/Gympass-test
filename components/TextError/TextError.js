import styled from 'styled-components';
import React from 'react';


let TextError = styled.span`
  color: red;
  font-size: 20px;
  margin-top: 20px;
  display: block;
`

export default TextError;

