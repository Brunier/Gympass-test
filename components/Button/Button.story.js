import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from './Button';

storiesOf('Button', module)
    .add('Normal', () => (
        <Button disabled={false} text={"Normal"}/>
    ))
    .add('Disabled', () => (
        <Button disabled={true} text={"Disabled"}/>
    ))
