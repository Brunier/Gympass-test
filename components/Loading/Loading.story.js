import React from 'react';
import { storiesOf } from '@storybook/react';
import Loading from './Loading';
import img from "../../static/loading.gif"

storiesOf('Loading', module)
    .add('Pequeno (20px)', () => (
        <Loading width={20} img={img} />
    ))
    .add('Medio (60px)', () => (
        <Loading width={60} img={img} />
    ))
    .add('Grande (100px)', () => (
        <Loading width={100} img={img} />
    ))
