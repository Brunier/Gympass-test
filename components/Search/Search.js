import React from 'react'
import styled from "styled-components";
import TextError from '../TextError/TextError';
import Loading from '../Loading/Loading';
import Button from '../Button/Button'

let Container = styled.div`
  margin-top:  ${props => props.marginTop ? props.marginTop + "px" : "1%"};
  text-align: center;
  color: #585656;
  
  input {
    padding: 10px 10px 10px 0;
    width: 15%;
    height: 27px;
    font-size: 23px;
    background-color: #fff;
    border: 2px solid #00757E;
    border-left: ${props => props.label ? "none" : "2px solid #00757E"};
    color: #585656;
  }
  
  label {
    background-color: white;    
    padding: 12px 7px 12px 12px;
    width: 110px;
    height: 23px;
    display: inline-block;
    font-size: 23px;
    border: 2px solid #00757E;
    border-right: ${props => props.label ? "none" : "2px solid #00757E"};
  }
`


class Search extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            item: null
        }
    }

    onFormSubmit = (e) =>  {
        e.preventDefault();
        this.props.onClick(this.state.item);
    };

    onChange = (event) => {
        this.setState({item: event.target.value})
        if(event.target.value === "" && this.props.onClean) {
            this.props.onClean();
        }
    }

    render() {
        return(
            <Container label={this.props.label} marginTop={this.props.marginTop}>
                <form onSubmit={this.onFormSubmit}>
                    <div>
                        {this.props.label ? <label>{this.props.label}</label> : ""}
                        <input data-cy={"search-input"} autoFocus placeholder={this.props.placeholder} onChange={this.onChange}/>
                        {
                            !this.props.loading ?
                                <Button type="submit" text={this.props.btnText} disabled={this.props.disabled !== undefined ? this.props.disabled : !this.state.item} />
                                : <Loading width={30} left={"2%"} display={"inline"} img={this.props.imgLoading}/>

                        }
                    </div>
                    {this.props.error ? <TextError data-cy={"search-error"}>{this.props.error}</TextError> : ""}
                </form>
               </Container>

        )
    }
}


export default Search;

