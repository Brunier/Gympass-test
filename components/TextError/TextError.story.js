import React from 'react';
import { storiesOf } from '@storybook/react';
import TextError from './TextError';


storiesOf('TextError', module)
    .add('Normal', () => (
        <TextError>Texto Error</TextError>
    ))
