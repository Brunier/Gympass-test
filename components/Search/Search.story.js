import React from 'react';
import { storiesOf } from '@storybook/react';
import Search from './Search';
import img from '../../static/loading.gif'


storiesOf('Search', module)
    .add('Normal', () => (
        <Search btnText={"Procurar Reposit贸rios"} placeholder={"Gympass"} />

))

    .add('Loading', () => (
        <Search  btnText={"Procurar Reposit贸rios"} placeholder={"Gympass"} loading={true} imgLoading={img}/>
    ))

    .add('Error', () => (
        <Search btnText={"Procurar Reposit贸rios"} placeholder={"Gympass"}  error={"Aconteceu um erro :("}/>
    ))

    .add('Com Label', () => (
        <Search label={"Github.com/"} btnText={"Procurar Reposit贸rios"} placeholder={"Gympass"}  error={"Aconteceu um erro :("}/>
    ))
