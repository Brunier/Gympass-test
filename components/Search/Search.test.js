import React from 'react';
import { shallow } from 'enzyme';
import TextError from '../TextError/TextError';
import Loading from '../Loading/Loading';
import renderer from 'react-test-renderer'
import Button from '../Button/Button'

// Components
import Search from './Search';

function setup(props) {
    const wrapper = shallow(<Search {...props} />);
    return { wrapper };
}

describe('Search Test Suite', () => {
    it('Com nome vazio, o botão tem que estar desabilitado', () => {
        const { wrapper } = setup();
        wrapper.setState({ item: '' });
        //expect(wrapper.find('button').exists()).toBe(true);
        expect(wrapper.find(Button).prop("disabled")).toBe(true);
    });
    it('Com nome, o botão tem que estar habilitado', () => {
        const { wrapper } = setup();
        wrapper.setState({ item: 'gympass' });
        //expect(wrapper.find('button').exists()).toBe(true);
        expect(wrapper.find(Button).prop("disabled")).toBe(false);
    });
    it('Botão desabilitado', () => {
        const { wrapper } = setup({disabled: true});
        //expect(wrapper.find('button').exists()).toBe(true);
        expect(wrapper.find(Button).prop("disabled")).toBe(true);
    });
    it('Botão habilitado', () => {
        const { wrapper } = setup({disabled: false});
        //expect(wrapper.find('button').exists()).toBe(true);
        expect(wrapper.find(Button).prop("disabled")).toBe(false);
    });
    it('Mostrar mensagem de erro', () => {
        const { wrapper } = setup({error: true});
        expect(wrapper.find(TextError).exists()).toBe(true);
    });
    it('Não mostrar a mensagem de erro', () => {
        const { wrapper } = setup({error: false});
        expect(wrapper.find(TextError).exists()).toBe(false);
    });
    it('Loading', () => {
        const { wrapper } = setup({loading: true});
        expect(wrapper.find(Loading).exists()).toBe(true);
    });
    it('Sem Loading', () => {
        const { wrapper } = setup({loading: false});
        expect(wrapper.find(Loading).exists()).toBe(false);
    });
    it('Snapshot', () => {
        const tree = renderer.create(<Search />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
