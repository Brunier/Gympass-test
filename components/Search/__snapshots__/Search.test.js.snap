// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Test Suite Snapshot 1`] = `
<div
  className="sc-bxivhb emWzRi"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      
      <input
        autoFocus={true}
        data-cy="search-input"
        onChange={[Function]}
      />
      <button
        className="sc-htpNat nPeio"
        data-cy="button"
        disabled={true}
        type="submit"
      />
    </div>
    
  </form>
</div>
`;
